lib/srv/forward/sshserver.go

```diff
@@ -1348,8 +1348,10 @@ func (s *Server) handleX11Forward(ctx context.Context, ch ssh.Channel, req *ssh.
 		return trace.AccessDenied("X11 forwarding request denied by server")
 	}
 
+	// Set up the server to handle x11 channel requests from the client. If the OpenSSH
+	// control master option is in use, then the server might already be handling such requests.
 	err = x11.ServeChannelRequests(ctx, s.remoteClient.Client, s.handleX11ChannelRequest)
-	if err != nil {
+	if err != nil && !errors.Is(err, x11.ErrX11ForwardChannelAlreadyOpen) {
 		return trace.Wrap(err)
 	}
```

Comment:
This would benefit from being extracted as an `ensureX11Forwarding` method on `*Server` for clarity's sake.

Why do we even need this? Couldn't we handle `x11` channels all the time?

Reply:
> Why do we even need this? Couldn't we handle x11 channels all the time?

I think you're right and I've changed it to always handle x11 channel requests, but it may be up for debate. 

Quoting the comment in my latest changes: 
```
According to RFC 4254, client "implementations MUST reject any X11 
channel open requests if they have not requested X11 forwarding". 
However, since this is a forwarding client implementation, we should 
simply accept and forward, leaving it up to the remote client to reject
the channel request.
```

Since client implementations must reject x11 channel requests, the forwarding client can safely assume that the channel request it forwards will be rejected.

However if we want to be abundantly cautious and follow the spec to a T, we could instead introduce a `Server.x11Requested` flag to determine whether we accept or reject x11 channel requests in `Server.handleX11ChannelRequest`. We'd then set that flag to true at the end of `Server.handleX11Forward`.